### ISSUE TITLE
Apparent flipping point of ViewCam

### ISSUE THREAD

**Issue by BalmyWinner** [GuyLivingUnderARock/Solid-Waddle-3D-Engine#2]
The cubes are 100x100x100 units and are each 100 units apart. The centre cube is at (0, 0, 0) in the world space.
The FOV angle used is 60 degrees.

At -750 z the camera's centre point inverts meshes and warps them towards the centre.
![image](https://github.com/GuyLivingUnderARock/Solid-Waddle-3D-Engine/assets/50259456/cf080324-df7e-49c0-be6e-51e8f8a697ae)

![image](https://github.com/GuyLivingUnderARock/Solid-Waddle-3D-Engine/assets/50259456/1dc001c5-2eb4-45d7-adb4-31540a62432d)

At anything smaller than -750 z the meshes stop travelling away from the camera and start returning but are flipped.
![image](https://github.com/GuyLivingUnderARock/Solid-Waddle-3D-Engine/assets/50259456/48d06b08-2e4c-41c3-8278-46e684703b79)

**Comment by BalmyWinner** [comment]
Also does this.
![image](https://github.com/GuyLivingUnderARock/Solid-Waddle-3D-Engine/assets/50259456/446e5b93-be5c-4aa5-a930-aebb46fa01f4)

**Comment by BalmyWinner** [comment]
Currently using this function to convert 3D points to 2D.
```
SDL_FPoint Point3D::ConvertToSDL_FPoint(float focalDistance, Point3D camPosition, Point3D camRotation) {
	SDL_FPoint point;

	float h_x = std::sqrt(std::pow(z + camPosition.z + focalDistance, 2) + std::pow(x - camPosition.x, 2));
	float h_y = std::sqrt(std::pow(z + camPosition.z + focalDistance, 2) + std::pow(y + camPosition.y, 2));

	float angle_x = std::asin((x - camPosition.x) / h_x);
	float angle_y = std::asin((y + camPosition.y) / h_y);

	point.x = h_x * std::sin(angle_x) * (z + camPosition.z + focalDistance) / focalDistance + WINDOW_WIDTH/2;
	point.y = h_y * std::sin(angle_y) * (z + camPosition.z + focalDistance) / focalDistance + WINDOW_HEIGHT/2;

	return point;
}
```

**Comment by BalmyWinner** [comment]
A potential fix might be to find a Pythagorean replacement for `std::sin(angle_x)` and `std::sin(angle_y)`.

**Comment by GuyLivingUnderARock** [comment]
Note: C++ cmath trig functions use radians not degrees

**Comment by BalmyWinner** [comment]
Fixed in commit aa59ea348e3bc6b1c6671e504d736053376b0a59
